R/plot_cluster_sina.R

```diff
@@ -24,17 +22,46 @@ plot_cluster_sina <- function(pldf,
     lineage_regexp = lineage_regexp
   )
 
-  p1 <- create_cluster_sina_ggplot(sc1, y_lab = varx)
+  create_cluster_sina_ggplot(sc1, y_lab = varx)
+}
 
-  g1 <- create_widget(
-    ggobj = p1,
-    width_svg = 8,
-    height_svg = 8
-  )
+#' Save a sina-cluster-plot to a file
+#'
+#' Saves as either an htmlwidget (in a \code{html} file) or a ggplot object (in a \code{rds} file).
+#'
+#' @param   ggobj   \code{ggplot2} object. Contains the plot that is to be saved.
+#' @param   varx   Scalar string. Which variable is depicted in the plot?
+#' @param   output_dir   File path. The directory where the plot will be stored.
+#' @param   output_format   Scalar string (either \code{rds} or \code{html}). In which format should
+#'   the plot be saved? Default: \code{rds}.
+#' @param   width_svg,height_svg   The width and height of the plot (only used when
+#'   \code{output_format == "html"}).
+#'
+#' @return   Invisibly returns the file path where the plot was saved
 
-  htmlwidgets::saveWidget(g1,
-    file = as.character(glue::glue("{output_dir}/sina-{varx}.html"))
-  )
+save_sina_plot <- function(ggobj,
+                           varx,
+                           output_dir,
+                           output_format = c("rds", "html"),
+                           width_svg = 8,
+                           height_svg = 8) {
+  output_format <- match.arg(output_format)
+
+  file_path <- file.path(output_dir, glue::glue("sina-{varx}.{output_format}"))
+
+  if (output_format == "rds") {
```

Comment:
Ah. Maybe not yet. If they have produced a cluster-sina ggplot object, then they can save that as an .rds or an .html by calling `save_sina_plot(ggobj, ..., output_format = "rds")` or `save_sina_plot(ggobj, ..., output_format = "html")`.

But a typical user would be calling the `treeview()` workflow, rather than the internals. I can add a `sina_output_format` parameter to `treeview()` so that they can call `treeview(input_env_rds, sina_output_format = "rds")` or `treeview(input_env_rds, sina_output_format = "html")`.

Will do that now.

Reply:
Sorry, my misunderstanding. As both "html" and "rds" in the same runthrough? That isn't implemented here, but can easily be done. Give me a minute

Reply:
This has been rewritten. Now the user can save cluster-sina plots as either .rds, .html or both (default: both). And they can specify which format to output the cluster-sina plots in the arguments to `treeview()`.

So `treeview(input_env_rds)` will output both .rds and .html versions of `sina-logistic_growth_rate` and `sina-clock_outlier` to "./treeview/".